### DISCUSSION TITLE
Idea: renovate should distinguish between closed-merged and closed-unmerged PRs when deciding what to recreate

### DISCUSSION THREAD

**Discussion by nyctef** [renovatebot/renovate#25833] [Suggest an Idea]
Overall renovate has been working great for us - thanks! We've been hitting a couple of related issues a few times, though:

- In one case, renovate updates some package from v3 -> v4, and everything is fine. But if later that package is installed at v3 into some other project, then renovate leaves it alone since there's already a closed (merged) PR for the v4 update. Then the repo gets stuck with some inconsistent versions if it's not fixed manually. Ideally this PR should get recreated, since the previous PR was closed and merged.
- In another case, we have the "[immortal PRs](https://docs.renovatebot.com/key-concepts/pull-requests/#immortal-prs)" problem where a PR for the `dotnet-monorepo` group keeps getting immediately recreated, even though we've closed it without merging. Ideally this PR should *not* get recreated, since the previous PR was closed without merging.
  - This is slightly suboptimal because it'll prevent all future updates to this group, even when new versions arrive. I assume we'd still be able to manually resume these updates via the dependency dashboard issue, though.
  - If we wanted to stop these updates long-term then we could change `renovate.json` to update them. But that's not ideal since we only want to temporarily pause non-major updates while we're working on the .NET 8 upgrade.

So tl;dr **I think it would be ideal if (by default) closed-merged PRs get recreated, but closed-unmerged PRs don't**. Does that make sense? Are there some unforeseen consequences that might make it not work? Is this already taken into account and I'm just not spotting it? :)

related issue: https://github.com/renovatebot/renovate/issues/22251 which I think proposes a different but related solution for the immortal PRs case

**Comment by rarkins** [comment]
Renovate should already differentiate between closed-merged and closed-unmerged mortal PRs (this wasn't always the behavior). Need more details such as logs if that isn't what you're seeing

**Reply by nyctef** [reply]
Ah, that sounds promising - the first case was remembered from a while ago, so maybe that works better now. I'll try and reproduce it again if I get some time 👍

**Reply by nyctef** [reply]
I've just tried to reproduce the issue we had in the first case and it does seem like that's fixed now, thanks! I think this might still be a useful idea for the second case (preventing certain immortal PRs from being reopened) but the fingerprinting idea is probably a better solution there anyway
